### ISSUE TITLE
Add jetstream domain option 

### ISSUE THREAD

**Issue by phho** [nats-io/nats.java#542]
nats start to support for multiple JetStream domains across Leafnodes from [nats server 2.2.3](https://github.com/nats-io/nats-server/releases/tag/v2.2.3). Please add jetstream domain option in nats.java to communicate with  JetStream's unique domain specific prefix [$JS.\<domain\>.API](https://docs.nats.io/jetstream/leafnodes). Thanks.

**Comment by scottf** [comment]
Right now you can set the entire prefix using the JetStreamOptions which can then be passed to the connection when create a JetStream context. We are considering adding another variation to JetStreamOptions which will take a domain and construct the prefix for you.

**Comment by phho** [comment]
Hi @scottf , 
Thanks for your quickly reply. I try to set jetstream domain prefix with JetStreamOptions class and has exception as below. (jnats:2.12.0) 
```java
# build JetStreamOptions
JetStreamOptions jsOptions = JetStreamOptions.builder().prefix("$JS.mydomain.API").build();
# Exception
Exception in thread "main" java.lang.IllegalArgumentException: Prefix must be in the printable ASCII range and cannot include `*`, `>` or `$` [$JS.mydomain.API]
	at io.nats.client.support.Validator.lambda$validatePrintableExceptWildGtDollar$5(Validator.java:144)
	at io.nats.client.support.Validator._validate(Validator.java:90)
	at io.nats.client.support.Validator.validatePrintableExceptWildGtDollar(Validator.java:142)
	at io.nats.client.support.Validator.validateJetStreamPrefix(Validator.java:46)
	at io.nats.client.support.JsPrefixManager.addPrefix(JsPrefixManager.java:36)
	at io.nats.client.JetStreamOptions$Builder.build(JetStreamOptions.java:146)
	at test.JetStreamPrefixTest.main(JetStreamPrefixTest.java:11)
```

**Comment by scottf** [comment]
Ok I'll look at that

**Comment by scottf** [comment]
This is being fixed in this PR https://github.com/nats-io/nats.java/pull/549

**Comment by phho** [comment]
Thank you for paying attention to the question and fixed quickly. Will the change be released in the next jnats version?

**Comment by scottf** [comment]
@phho Yes, it's merged to the main branch, so it should be available as a snapshot 2.12.1 See the Readme for more information on snapshots https://github.com/nats-io/nats.java#installation
